controller/src/main/java/io/pravega/controller/store/PravegaTablesStoreHelper.java

```diff
@@ -244,6 +249,89 @@ public void invalidateCache(String table, String key) {
                 .whenComplete((r, ex) -> releaseEntries(entries));
     }
 
+    public CompletableFuture<List<Version>> removeValues(final String tableName, Set<String> tableKeys, final String appendValue, long requestId) {
+        Supplier<String> errorMessage = () -> String.format("remove values: on table: %s", tableName);
+
+        // read values and add values
+        return withRetries(() -> removeValuesTables(tableName, appendValue, tableKeys, requestId)
+                .thenApplyAsync(x -> {
+                    log.trace("appendValues {} updated to table {} with version {}", appendValue, tableName, x);
+                    return x.stream().map(v -> (Version) new Version.LongVersion(v.getSegmentVersion())).collect(Collectors.toList());
+                }), errorMessage, true, requestId);
+    }
+
+
+    public CompletableFuture<List<Version>> appendValues(final String tableName, Set<String> tableKeys, final String appendValue, long requestId) {
```

Comment:
same here, please add the java doc.. 
do we need this method in this class or is this business logic very specific to the tags and should be moved there?

i believe you are appending a new value to the end of existing values.. where this method would do a get and put. right?
this is because this method is implicitly assuming that the value is a List<Strings> to which it needs to append. 
if we want to create a helper method of this sort, it should use generics for the value and clearly call out that it works when the value is actually serialized as a list. 

Ideally i would want to move this logic to either the caller or some helper class for that business logic if it has to be reused. it wouldnt be a natural fit for the storehelper which is generic for all types of values. only thing it fixes is that keys be strings.

Reply:
improved the code here. Let me know if it can be improved further.